week1/lc378_kthSmallest.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @param {number[][]} matrix
+ * @param {number} k
+ * @return {number}
+ */
+const kthSmallest = function(matrix, k) {
+    let start = matrix[0][0];
+    let end = matrix[matrix.length - 1][matrix.length - 1];
+
+    while (start + 1 < end) {
+        let mid = start + ((end - start) >> 1);
+
+        if (countLessAndEqualT(matrix, mid) < k) {
+            start = mid;
+        }
+        else {
+            end = mid;
+        }
+    }
+    return (countLessAndEqualT(matrix, start) >= k ? start : end);
+};
+
```

Comment:
We can also start counting from top right corner of the matrix.
` const countLessAndEqualT = function (matrix, t) {
    let counter = 0;
    let row = 0;
    let col = matrix.length - 1;
     
    while (row < matrix.length && col >= 0) {
        if (matrix[row][col] <= t) {
            counter = counter + (col + 1);
            row++;
        } else {
            col--;
        }
    }
    return counter;
 }; `

Reply:
Are you sure?

Reply:
@Houjue92 How do you think?